nsdb-cluster/src/main/scala/io/radicalbit/nsdb/cluster/logic/WriteConsistencyLogic.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018-2020 Radicalbit S.r.l.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.radicalbit.nsdb.cluster.logic
+
+import akka.actor.Actor
+import akka.cluster.ddata.Replicator.{WriteAll, WriteConsistency, WriteMajority}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+trait WriteConsistencyLogic { this: Actor =>
+  private val config = context.system.settings.config
+
+  private val writeDuration = 5.seconds
+
+  protected val writeConsistency: WriteConsistency =
+    WriteConsistencyLogic.fromConfigValue(config.getString("nsdb.cluster.write-consistency"))(writeDuration)
+}
+
+object WriteConsistencyLogic {
```

Comment:
we can avoid to define this object since it contains only one method and move that mothod logic into the lazy val definition

Reply:
also, since we are inside a trait this must be lazy